opm/simulators/wells/MultisegmentWell_impl.hpp

```diff
@@ -2268,11 +2243,81 @@ namespace Opm
             ++count;
         }
 
+        const double control_tolerance = getControlTolerance(deferred_logger);
+        if (residuals[SPres + 1] > control_tolerance) {
+            sum += residuals[SPres + 1] / control_tolerance;
+            ++count;
+        }
+
         // if (count == 0), it should be converged.
         assert(count != 0);
 
-        // return sum / double(count);
         return sum;
     }
 
+
+
+
+
+    template<typename TypeTag>
+    double
+    MultisegmentWell<TypeTag>::
+    getControlTolerance(DeferredLogger& deferred_logger) const
+    {
+        double control_tolerance = 0.;
+        switch(well_controls_get_current_type(well_controls_) ) {
+            case BHP:
```

Comment:
In the master branch, under BHP control, we will also use `param_.tolerance_wells_` for the BHP control check. It looks for the BHP control equation to keep a tighter tolerance (`param_.tolerance_wells_`) is good for the general iteration.

I am re-running all the tests with the tighter tolerance.

Reply:
It fixes the performance regression.  There is one case the performance encounters regression in the master branch during past about 7 days, not due to this PR.